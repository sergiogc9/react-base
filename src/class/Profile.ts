export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
