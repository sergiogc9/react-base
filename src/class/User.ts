export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
