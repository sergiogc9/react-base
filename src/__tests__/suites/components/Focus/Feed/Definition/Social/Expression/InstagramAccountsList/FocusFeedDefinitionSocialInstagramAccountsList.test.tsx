import React from "react";
import { shallow } from "enzyme";

import FocusFeedDefinitionSocialInstagramAccountsList from "@src/components/Focus/Feed/Definition/Social/Expression/InstagramAccountsList/FocusFeedDefinitionSocialInstagramAccountsList";
import { SocialInstagramAccount, DefinitionSocialInstagramAccount } from "@src/class/Feed";

const instagramAccount: SocialInstagramAccount = {
	id: "id1",
	name: "Girona FC",
	profile_picture_url: "fakeurl.com",
	screen_name: "gironafc",
	valid: true
};
const instagramAccount2: SocialInstagramAccount = {
	id: "id2",
	name: "Fake Instagramer",
	profile_picture_url: "fakeurl.com",
	screen_name: "fake_instagramer",
	valid: true
};
const instagramAccounts = [instagramAccount, instagramAccount2];
const definitionInstagramAccount: DefinitionSocialInstagramAccount = {
	id: "id1",
	screen_name: "gironafc",
	linkedExpression: { type: "include_expressions", index: 0 }
};

describe("Definition social instagram accounts list", () => {
	let wrapper: any;
	let onInstagramAccountSelected = jest.fn();

	const getComponent = () => (
		<FocusFeedDefinitionSocialInstagramAccountsList
			instagramAccounts={instagramAccounts}
			definitionInstagramAccounts={[]}
			expressionKey="include_expressions"
			index={0}
			onInstagramAccountSelected={onInstagramAccountSelected}
		/>
	);

	beforeEach(() => {
		onInstagramAccountSelected = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component default snaphsot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snaphsot without instagram accounts loaded", () => {
		wrapper.setProps({ instagramAccounts: null });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snaphsot with an instagram account used in another expression", () => {
		wrapper.setProps({ definitionInstagramAccounts: [definitionInstagramAccount] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snaphsot with dialog shown", () => {
		wrapper.setState({ showFacebookDialog: true });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Facebook dialog onClose handler", () => {
		wrapper.setState({ showFacebookDialog: true });
		wrapper.find('Connect(FocusFeedDefinitionSocialFacebookDialog)').at(0).props().onClose();
		expect(wrapper.state().showFacebookDialog).toBe(false);
	});

	it("On click connect new account should show facebook dialog", () => {
		wrapper.find('#focusFeedDefinitionSocialInstagramAccountAddNew').at(0).simulate('click');
		expect(wrapper.state().showFacebookDialog).toBe(true);
	});

	it("On instagram account clicked", () => {
		expect(onInstagramAccountSelected).toHaveBeenCalledTimes(0);
		wrapper.find(`FocusFeedDefinitionSocialInstagramAccount`).at(0).props().onSelected();
		expect(onInstagramAccountSelected).toHaveBeenCalledTimes(1);
		expect(onInstagramAccountSelected).toHaveBeenCalledWith({
			id: definitionInstagramAccount.id,
			screen_name: definitionInstagramAccount.screen_name,
			linkedExpression: { type: "include_expressions", index: 0 }
		});
	});
});
