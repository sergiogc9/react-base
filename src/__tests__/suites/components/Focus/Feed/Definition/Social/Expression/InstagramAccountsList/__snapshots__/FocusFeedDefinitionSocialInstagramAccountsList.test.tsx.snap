// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition social instagram accounts list Component default snaphsot 1`] = `"<div id=\\"focusFeedSocialDefinitionInstagramAccountsinclude_expressions0\\" class=\\"social-definition-instagram-accounts-list\\"><div id=\\"focusFeedDefinitionSocialInstagramAccountid1\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Girona FC</span><span>(@gironafc)</span></div><div id=\\"focusFeedDefinitionSocialInstagramAccountid2\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Fake Instagramer</span><span>(@fake_instagramer)</span></div><div id=\\"focusFeedDefinitionSocialInstagramAccountAddNew\\" class=\\"instagram-account-add\\"><i class=\\"md-icon material-icons\\">add</i><span>feed.form.instagram.legend.button_react</span></div></div>"`;

exports[`Definition social instagram accounts list Component snaphsot with an instagram account used in another expression 1`] = `"<div id=\\"focusFeedSocialDefinitionInstagramAccountsinclude_expressions0\\" class=\\"social-definition-instagram-accounts-list\\"><div id=\\"focusFeedDefinitionSocialInstagramAccountid2\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Fake Instagramer</span><span>(@fake_instagramer)</span></div><div id=\\"focusFeedDefinitionSocialInstagramAccountAddNew\\" class=\\"instagram-account-add\\"><i class=\\"md-icon material-icons\\">add</i><span>feed.form.instagram.legend.button_react</span></div></div>"`;

exports[`Definition social instagram accounts list Component snaphsot with dialog shown 1`] = `
"<div id=\\"focusFeedSocialDefinitionInstagramAccountsinclude_expressions0\\" className=\\"social-definition-instagram-accounts-list\\">
  <Connect(FocusFeedDefinitionSocialFacebookDialog) onClose={[Function: onClose]} />
  <FocusFeedDefinitionSocialInstagramAccount account={{...}} removable={false} onSelected={[Function: onSelected]} t={[Function: t]} />
  <FocusFeedDefinitionSocialInstagramAccount account={{...}} removable={false} onSelected={[Function: onSelected]} t={[Function: t]} />
  <div id=\\"focusFeedDefinitionSocialInstagramAccountAddNew\\" className=\\"instagram-account-add\\" onClick={[Function: onClick]}>
    <FontIcon iconClassName=\\"material-icons\\">
      add
    </FontIcon>
    <span>
      feed.form.instagram.legend.button_react
    </span>
  </div>
</div>"
`;

exports[`Definition social instagram accounts list Component snaphsot without instagram accounts loaded 1`] = `"<div id=\\"focusFeedSocialDefinitionInstagramAccountsinclude_expressions0\\" class=\\"social-definition-instagram-accounts-list\\"><svg id=\\"focusFeedSocialDefinitionInstagramAccountsLoadinginclude_expressions0\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate\\" width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div>"`;
